plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryMappingFileProvider.kt

```diff
@@ -0,0 +1,30 @@
+package io.sentry.android.gradle
+
+import com.android.build.gradle.api.ApplicationVariant
+import org.gradle.api.Project
+import java.io.File
+
+internal object SentryMappingFileProvider {
+
+    /**
+     * Returns the obfuscation mapping file (might be null if the obfuscation is disabled).
+     *
+     * @return the mapping file or null if not found
+     */
+    @JvmStatic
+    fun getMappingFile(project: Project, variant: ApplicationVariant): File? =
+        try {
+            val mappingFiles = variant.mappingFileProvider.get().files
+            if (mappingFiles.isEmpty()) {
+                project.logger.warn("[sentry] .mappingFileProvider.files is empty for ${variant.name}")
+                null
+            } else {
+                project.logger.info("[sentry] Mapping File ${mappingFiles.first()} for ${variant.name}")
+                mappingFiles.first()
+            }
+        } catch (ignored: Throwable) {
+            project.logger.error("[sentry] .mappingFileProvider failed with: ${ignored.message}")
+            ignored.printStackTrace()
+            variant.mappingFile
```

Comment:
in theory we don't need to support that anymore

Reply:
Yup but then the AGP matrix will fail for 3.x. I'd say let's remove the 3.x support after the release of plugin v2

Reply:
yeah just have seen the CI.
so either we keep support for AGP 3.x or we drop `before` the release of Plugin v2, as we are doing a major bump, it's fine to do such breaking change, but after releasing would require another major bump